src/Features/JsonPatch.SystemTextJson/src/Helpers/JsonUtilities.cs

```diff
@@ -10,8 +11,41 @@ internal static class JsonUtilities
 {
     public static bool DeepEquals(object a, object b, JsonSerializerOptions serializerOptions)
     {
-        return JsonObject.DeepEquals(
-            JsonSerializer.SerializeToNode(a, serializerOptions),
-            JsonSerializer.SerializeToNode(b, serializerOptions));
+        if (a == null && b == null)
+        {
+            return true;
+        }
+
+        if (a == null || b == null)
```

Comment:
Is it possible for one of the inputs to be a JsonElement representing `null` while the other is just CLR `null`?

Reply:
Not sure, theoretically - maybe. I'd prefer to wait for actual feedback on this one from customers before trying to expand the implementation to cover that scenario too.